### ISSUE TITLE
Inconsistent calculations in CategoricalCrossentropy using logits

### ISSUE THREAD

**Issue by Frightera** [keras-team/tf-keras#242]
**Describe the problem**.

It seems like a bug in the calculation of cross entropy while calculating the log.

**Describe the current behavior**.
If the logits get too big, the calculation of `CategoricalCrossEntropy` is effected. We expect two of them to be same:

```python
logits = tf.constant([[14.4, 10.1, 3.5]])
y_true = tf.constant([[1, 0, 0]], dtype=tf.float32)

cce_obj = keras.losses.CategoricalCrossentropy(
                      from_logits=True)(y_true, logits)

cce_obj_logits = keras.losses.CategoricalCrossentropy(
                      from_logits=False)(y_true, tf.nn.softmax(logits))
```

which returns the same value, `0.01349`.

If the logits get bigger, they will return different values.

**Describe the expected behavior**.
Those two losses should produce the same result.

- Do you want to contribute a PR? (yes/no): Yes. Thim seems like an issue while calculating the log in categorical cross entropy losses.

**Standalone code to reproduce the issue**.
Please check the [gist here](https://colab.research.google.com/gist/Frightera/3959f54c0d06d69b0683161032c29c4f/cce-inconsistent-results.ipynb
). 

Edit (13/03/2023): 

Here are the results when logits are _larger_:

```python
logits = tf.constant([[14.4, 50.1, 3.5]], dtype = tf.float64)
y_true = tf.constant([[1, 0, 0]], dtype=tf.float64)

cce_obj = keras.losses.CategoricalCrossentropy(
                      from_logits=True)

loss = cce_obj(y_true, logits)
loss.numpy() # 35.7

cce_obj = keras.losses.CategoricalCrossentropy(
                      from_logits=False)

loss = cce_obj(y_true, tf.nn.softmax(logits))
loss.numpy() # 16.11
```
